events/models.py

```diff
@@ -60,13 +60,30 @@ def clean(self):
                 raise ValidationError(
                     "For a booking today, the time must be in the future")
 
+        # If we are updating the event, we cannot modify the initial/max size
+        # if that modification would fill the event based on already made
+        # reservations.
+        reservation = Reservation.objects.filter(event=self, user=self.host).first()
+        if reservation:
+            if (self.reserved_places() - reservation.places + self.initial_size > self.max_size):
+                if (self.initial_size <= reservation.places):
+                    raise ValidationError("Cannot reduce the size of the event as reservations have already filled the event.")
+                else:
+                    raise ValidationError("Cannot increase the number of initial players as reservations have already filled the event.")
+
     def save(self, *args, **kwargs):
         self.clean()
         super(Event, self).save(*args, **kwargs)
         # Create the event with an initial reservation for the host
         if self.initial_size > 0:
-            Reservation.objects.create(
-                user=self.host, event=self, places=self.initial_size)
+            reservation = Reservation.objects.filter(user=self.host, event=self).first()
```

Comment:
Instead of trying to find a reservation object with the host to determine whether the event is being updated or not, you can alternative use `if not self.pk` (true if it's a new object). I think this is a bit cleaner. When the object is saved to the data base this value (primary key) will be created and become truthy.